clients/pkg/logentry/stages/limit.go

```diff
@@ -88,3 +135,9 @@ func (m *limitStage) shouldThrottle() bool {
 func (m *limitStage) Name() string {
 	return StageTypeLimit
 }
+
+func getDropCountByLabelMetric(registerer prometheus.Registerer) *prometheus.CounterVec {
+	return util.RegisterCounterVec(registerer, "logentry", "dropped_lines_by_label_total",
+		"A count of all log lines dropped as a result of a pipeline stage",
+		[]string{"label_name", "label_value"})
```

Comment:
ratelimiting itself is a safeguard against misbehaving apps. Expectation is that in the common case ratelimit wont be hit, but if it is hit you'd want to  know which app is being ratelimited, hence we report both label name and value.

In case too many label values are being ratelimited (and hence their counters created) on a grossly misconfigured setup, we cap memory usage by `Reset()` counter every time we GC to maintain maximum configured capacity.

Reply:
I see a lot of cases when we might continue reporting thousands of the metrics, (for each label value) ...
for example: 
user set `MaxDistinctLabels` to 100_000.
90_000 of streams are got rate limited, so we started reporting 90_000 metrics... and we will continue reporting these metrics forever if we do not reach `MaxDistinctLabels`... until we call `gcCb`.
Moreover, if you have 100 instances of promtail, and all of them report 90_000 metrics you will have 9_000_000 metrics ...
I also think that it would be useful to see this metric, but we can not do it by adding such unique values as labels of the metrics...

Reply:
In general having high cardinality label in Loki is a nogo anyway. Also reporting stale metrics potentially forever is a long standing problem with Prometheus instrumentation and whole ecosystem learned to live with it. 

Label value cardinality is specific to every use case, but in general it is expected that:
- low cardinality labels are used 
- ratelimits are set so that in normal operation mode they are not triggered

For scenario you worry about both have be false and even then worst thing happening is just 1 extra metric per label.

As an administrator of log shipping I need to know which teams to contact to tame their log usage and this metric gives me this visibility. In the absence of per label metric  all I have is global drop count and it wont be enough.

Reply:
completely agree with you ;) it is super useful to have this metric with label_value.... and considering that in normal operation mode it should not be a case, then I agree that we can leave it ;)